src/tools/piv-tool.c

```diff
@@ -284,14 +283,16 @@ static int admin_mode(const char* admin_info)
 /* generate a new key pair, and save public key in newkey */
 static int gen_key(const char * key_info)
 {
-	int r;
+	int r = -1;
```

Comment:
This is a little odd: `main()` directly takes the error code from `gen_key()` and outputs it if it is non-zero. From this, I would expect `gen_key()` to return a positive number. Other internal functions that are called from `main()` are restricting themselves to small positive values... I think the refactoring done here would give a good opportunity to fix the return value from `gen_key()`. Do you agree, @dengert ?

Reply:
Right. It was mess already and I did not want to mess with it too much. But looks like I already did so lets fix this a bit.